cmd/apidemic/main.go

```diff
@@ -38,8 +50,66 @@ func main() {
 					Value:  3000,
 					EnvVar: "PORT",
 				},
+				cli.StringFlag{
+					Name:   "endpoint-dir",
+					Usage:  "Directory to scan for endpoint",
+					Value:  "./endpoints/",
+					EnvVar: "ENDPOINT_DIR",
+				},
 			},
 		},
 	}
 	app.RunAndExitOnError()
 }
+
+// addEndpoints iterates over a directory and tries to register each file as an endpoint
+// if directory does not exist or not readable it will simply return without registration
+// failing registration however causes an error response
+func addEndpoints(s *mux.Router, endpointDir string) error {
+	var registerPayload map[string]interface{}
+	files, err := ioutil.ReadDir(endpointDir)
+	if err != nil {
+		return nil
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		filePath := endpointDir + file.Name()
+
+		registerPayload, err = getRegisterPayload(filePath)
+		if err != nil {
+			return err
+		}
+
+		w := httptest.NewRecorder()
+		req := apidemic.JsonRequest("POST", "/register", registerPayload)
+		s.ServeHTTP(w, req)
+
+		log.Printf("%s is registered\n", filePath)
```

Comment:
No need, for this. It just adds noise.

Reply:
Actually I found this very useful and only adds noice at the beginning. Maybe I can add a verbose flag to see this log?

Reply:
I see, you can leave this for now.